src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -0,0 +1,115 @@
+@WebServlet("/search")
+public class SearchServlet extends HttpServlet {
+
+  @Override
+  /* Show classes. */
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<String> classNames = new ArrayList<>();
+    List<String> professorNames = new ArrayList<>();
+    List<Filter> filters = new ArrayList<>();
+    if (!request.getParameter("className").isEmpty()) {
+      String className = request.getParameter("className");
+      Filter classFilter = new FilterPredicate("Name", FilterOperator.EQUAL, className);
+      filters.add(classFilter);
+    }
+    if (!request.getParameter("profName").isEmpty()) {
+      String profName = request.getParameter("profName");
+      Filter profFilter = new FilterPredicate("Professor", FilterOperator.EQUAL, profName);
+      filters.add(profFilter);
+    }
+    List<Integer> units = new ArrayList<>();
+    if (!request.getParameter("units").isEmpty()) {
+      List<String> strUnits = Arrays.asList(request.getParameter("units").split(","));
+      for (String number : strUnits) {
+        units.add(Integer.valueOf(number));
+      }
+      if (units.size() != 0) {
+        Filter unitsFilter = new FilterPredicate("Units", FilterOperator.IN, units);
+        filters.add(unitsFilter);
+      }
+    }
+
+    Query q = new Query("Class");
+    // Add filters depending on number of fields specified
+    if (filters.size() == 1) {
+      q.setFilter(filters.get(0));
+    } else if (filters.size() == 2) {
+      q.setFilter(CompositeFilterOperator.and(filters.get(0), filters.get(1)));
+    } else if (filters.size() == 3) {
+      q.setFilter(
+          CompositeFilterOperator.and(
+              CompositeFilterOperator.and(filters.get(0), filters.get(1), filters.get(2))));
+    }
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    List<Entity> classes = datastore.prepare(q).asList(FetchOptions.Builder.withDefaults());
+    for (Entity entity : classes) {
+      String name = (String) entity.getProperty("Name");
+      classNames.add(name);
+      String professor = (String) entity.getProperty("Professor");
+      professorNames.add(professor);
+    }
+
+    // Generate links for class pages and send them back - set variables in URL
+    String classNameJson = new Gson().toJson(classNames);
+    String profNameJson = new Gson().toJson(professorNames);
+    JSONObject obj = new JSONObject();
+    obj.put("classNames", classNameJson);
+    obj.put("profNames", profNameJson);
+    response.setContentType("application/json;");
+    response.getWriter().println(obj);
+  }
+
+  /* Store class. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = request.getParameter("class-name");
+    String prof = request.getParameter("prof-name");
+    Integer units = Integer.valueOf(request.getParameter("num-units"));
```

Comment:
This is okay for now but in the future you'll have to think about how to handle a case where a non-integer is returned.

Reply:
Just made this feature a radio button to avoid this issue. I had check boxes for the search feature but just hadn't gotten around to mirroring that in the input feature yet since I wasn't sure if Dagm had done something similar yet. Thanks!

Reply:
That's true but people can always query your server directly without using your client : )

Reply:
Ooh good point - didn't think about that

Reply:
I'll add in error checking and testing later this week most likely - just wanna get search working and robus for the proper inputs first